src/routes/profile.js

```diff
@@ -155,30 +193,139 @@ router.get(
   },
 );
 
+const storage = multer.memoryStorage();
+const upload = multer({ storage: storage }).fields([
+  { name: 'avatarRes', maxCount: 1 },
+  { name: 'avatarSpy', maxCount: 1 },
+]);
+
 // Update the customavatar
 router.post(
   '/:profileUsername/changeavatar',
   checkProfileOwnership,
-  (req, res) => {
-    console.log('Recieved change avatar');
-    console.log(`For user ${req.params.profileUsername}`);
-    console.log(`Res link: ${req.body.reslink}`);
-    console.log(`Spy link: ${req.body.spylink}`);
-    console.log(`Message to mod: ${req.body.msgToMod}`);
+  upload,
+  async (req, res) => {
+    const MAX_AVATAR_REQUESTS = 2;
+    const MIN_GAMES_REQUIRED = 100;
+    const VALID_DIMENSIONS = [128, 1024];
 
-    // sometimes https links dont show up correctly
-    // req.body.reslink.replace("https", "http");
-    // req.body.spylink.replace("https", "http");
+    const s3 = req.s3;
+    const user = await User.findOne({ username: req.params.profileUsername });
 
-    const avatarRequestData = {
-      forUsername: req.params.profileUsername.toLowerCase(),
+    // TODO: how should i handle this?
+    if (!user) {
+      console.log('User not found');
+    }
 
-      resLink: sanitizeHtml(req.body.reslink),
-      spyLink: sanitizeHtml(req.body.spylink),
-      msgToMod: sanitizeHtml(req.body.msgToMod),
+    // Checks if custom avatar request is valid
+    if (user.totalGamesPlayed < MIN_GAMES_REQUIRED) {
+      req.flash(
+        'error',
+        `You must play at least 100 games to submit a custom avatar request. You have played ${user.totalGamesPlayed} games.`,
+      );
+      return res.redirect(
+        `/profile/${req.params.profileUsername}/changeavatar`,
+      );
+    }
 
-      dateRequested: new Date(),
+    let totalAvatarRequests = await avatarRequest.aggregate([
+      {
+        $match: {
+          forUsername: user.username.toLowerCase(),
+          processed: false,
+        },
+      },
+      {
+        $count: 'total',
+      },
+    ]);
 
+    totalAvatarRequests =
+      totalAvatarRequests.length === 0 ? 0 : totalAvatarRequests[0].total;
+
+    if (totalAvatarRequests >= MAX_AVATAR_REQUESTS) {
+      req.flash(
+        'error',
+        `You cannot submit more than ${MAX_AVATAR_REQUESTS} custom avatar requests.`,
+      );
+      return res.redirect(
+        `/profile/${req.params.profileUsername}/changeavatar`,
+      );
+    }
+
+    // TODO: should i check existence of req.files['avatarRes'][0] as well?
+    if (!req.files['avatarRes'] || !req.files['avatarSpy']) {
+      req.flash('error', 'You must submit a Res and Spy avatar.');
+      return res.redirect(
+        `/profile/${req.params.profileUsername}/changeavatar`,
+      );
+    }
```

Comment:
Another if statement for another early return.

Reply:
What did you mean by this one?